### ISSUE TITLE
String measuring is slow

### ISSUE THREAD

**Issue by Choco31415** [tjvr/scratchblocks#167]
Earlier I got to test the Scratchblocks v3 plugin on the international Test Wiki. MartinWollenweber and I tried it on this page: http://test.scratch-wiki.info/wiki/User:Mtwoll/s2

The page currently uses v2 of the Scratchblocks plugin, and the page renders all tags in less then a second. However, when I was testing v3, the page was taking over a minute to fully render. See:

http://i.cubeupload.com/Nie9NC.png
http://i.cubeupload.com/jItp3L.png

Basically, the plugin would render one tag a second.

**Comment by tjvr** [comment]
CPU profile:
<img width="380" alt="screen shot 2016-05-10 at 00 11 17" src="https://cloud.githubusercontent.com/assets/639289/15130931/eb29b03a-1643-11e6-8a3b-919f63214551.png">

**Comment by tjvr** [comment]
Oops. It turns out we create a hidden SVG for measuring strings for *each* call to `render()`. This is undesirable.

In addition, we could (and probably should) cache the strings.

**Comment by tjvr** [comment]
@Choco31415 I wrote a patch for this. Have a look at #168 and let me know if that solves the problem (ie. if that speeds it up).

**Comment by Choco31415** [comment]
I will check it out tomorrow.

**Comment by Choco31415** [comment]
I have confirmed that it works much faster now. There are about two seconds between when the page loads and when the blocks render.
